runcommandonset/tests/runcommandonset.set.tests.ps1

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'tests for runcommandonset set' {
+    BeforeAll {
+        $oldPath = $env:DSC_RESOURCE_PATH
+        $env:DSC_RESOURCE_PATH = Join-Path $PSScriptRoot ".."
+    }
+
+    AfterEach {
+        if (Test-Path $TestDrive/output.txt) {
+            Remove-Item -Path $TestDrive/output.txt
+        }
+    }
+
+    AfterAll {
+        $env:DSC_RESOURCE_PATH = $oldPath
+    }
+
+    It 'Input for executable and arguments can be sent to the resource' {
+        $input_json = @"
+        {
+            "executable": "pwsh",
+            "arguments": ["-Command", "echo hello world"]
+        }
+"@
+        $input_json | dsc resource set -r Microsoft.DSC.Transitional/RunCommandOnSet
+        # TODO: test output once DSC PR to capture it is merged
+        $LASTEXITCODE | Should -Be 0
+    }
+
+    It 'STDOUT captured via STDERR when calling resource directly' {
+        $input_json = @"
+        {
+            "executable": "pwsh",
+            "arguments": ["-Command", "echo hello world"]
+        }
+"@
+        $input_json | runcommandonset set 2> $TestDrive/output.txt
```

Comment:
Change all redirection to file to just assigning to variable

Reply:
Can I capture `STDERR` when it's assigned to a variable? I'm making sure the expected `STDOUT` and `STDERR` are written there

Reply:
`stderr` would still need to be redirected unfortunately.  There's a feature Jim was writing to redirect to a variable, but it doesn't exist yet